### PR TITLE
[Oresme] - Updates/fixes.

### PR DESCRIPTION
**Pull request by ivandrofly**

### CODE REVIEW

File: libse/SubtitleFormats/Oresme.cs [hunk 1]
```diff
@@ -90,24 +82,28 @@ public override void LoadSubtitle(Subtitle subtitle, List<string> lines, string
                     _errorCount++;
                 }
 
-                for (int i = 0; i < subtitle.Paragraphs.Count - 1; i++)
```
Comment (ivandrofly) [inline libse/SubtitleFormats/Oresme.cs]: must be outside reading loop....

File: libse/SubtitleFormats/Oresme.cs [hunk 2]
```diff
@@ -63,20 +47,28 @@ private static string EncodeTimeCode(TimeCode time)
...
-
+            Func<string, double> OptimialTimeProvider = Utilities.GetOptimalDisplayMilliseconds;
             subtitle.Paragraphs.Clear();
+            var sb = new StringBuilder(50);
             char[] splitChars = { '.', ':' };
             foreach (string line in lines)
             {
                 string s = line.Trim();
+                // invalid line
+                if (s.Length < 12)
```
Comment (ivandrofly) [inline libse/SubtitleFormats/Oresme.cs]: only try matching regex if length length >= 12

File: libse/SubtitleFormats/Oresme.cs [hunk 3]
```diff
@@ -13,29 +13,13 @@ public class Oresme : SubtitleFormat
...
+        public override string Name => "Oresme";
+        public override bool IsTimeBased => true;
 
         public override bool IsMine(List<string> lines, string fileName)
         {
             var subtitle = new Subtitle();
-
-            var sb = new StringBuilder();
-            foreach (string line in lines)
-                sb.AppendLine(line);
```
Comment (ivandrofly) [inline libse/SubtitleFormats/Oresme.cs]: futile

File: libse/SubtitleFormats/Oresme.cs [hunk 4]
```diff
@@ -90,24 +82,28 @@ public override void LoadSubtitle(Subtitle subtitle, List<string> lines, string
...
+            {
+                subtitle.Renumber();
+            }
         }
 
-        private static string GetText(string s)
+        private static string GetText(string s, StringBuilder sb)
         {
             s = s.Replace("{N}", Environment.NewLine);
-            var sb = new StringBuilder();
```
Comment (ivandrofly) [inline libse/SubtitleFormats/Oresme.cs]: avoid creating n* instance of StringBuilder...

**Comment by niksedk** [comment]
If this is still relevant please add a new pr with info about why, what and unit tests

**Pull request closed without merging**